### ISSUE TITLE
cache-dir in configfiile should be relative to that configfile

### ISSUE THREAD

**Issue by woutervh** [astral-sh/ruff#7958]
using ruff 0.0.292

in my projects's pyproject.toml, I configure the _cache-dir_

```
[tool.ruff]
# ruff searches for settings in the target-dir and up
# https://beta.ruff.rs/docs/configuration/
cache-dir = "var/cache/ruff"  # relative to $PWD
```

when setting a relative path in pyproject.toml,
 the setting should be relative to that pyproject.toml-file.

however, ruff takes it relative to the $PWD,  which does not make sense in a configfile.

when I run ruff from my project-directory,  it creates cache-dirs in every location I run ruff from:

```
pyproject.toml

docs/
    conf.py
    var/cache/ruff/
src/
    foo/
         var/cache/ruff
        __init__

tests/
    conftest.py
    var/cache/ruff/

var/cache/ruff/
```

**Comment by charliermarsh** [comment]
I don't think it should be relative to the configuration file in all cases (though it's probably not worth going into why), but it should be relative to the "project root" which is _typically_ the directory containing the configuration file.

In the above case, we _should_ be choosing the top-level directory as the project root in all cases, and so using that same top-level `var/cache/ruff/`, so seems like a bug.
